src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -165,111 +167,130 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
+        | TransactionGasParamAmounts
+        | LegacyTransactionGasParamAmounts;
 
-    let gasParams: TransactionGasParamAmounts | LegacyTransactionGasParamAmounts = {} as
-      | TransactionGasParamAmounts
-      | LegacyTransactionGasParamAmounts;
+      if (selectedGas.isEIP1559) {
+        gasParams = {
+          maxFeePerGas: selectedGas.maxBaseFee,
+          maxPriorityFeePerGas: selectedGas.maxPriorityFee,
+        };
+      } else {
+        gasParams = {
+          gasPrice: selectedGas.gasPrice,
+        };
+      }
 
-    if (selectedGas.isEIP1559) {
-      gasParams = {
-        maxFeePerGas: selectedGas.maxBaseFee,
-        maxPriorityFeePerGas: selectedGas.maxPriorityFee,
-      };
-    } else {
-      gasParams = {
-        gasPrice: selectedGas.gasPrice,
-      };
-    }
+      const { errorMessage } = await walletExecuteRap(wallet, type, {
+        ...parameters,
+        gasParams,
+        // @ts-expect-error - collision between old gas types and new
+        gasFeeParamsBySpeed: gasFeeParamsBySpeed,
+      });
+      isSwapping.value = false;
 
-    const { errorMessage } = await walletExecuteRap(wallet, type, {
-      ...parameters,
-      gasParams,
-      // @ts-expect-error - collision between old gas types and new
-      gasFeeParamsBySpeed: gasFeeParamsBySpeed,
-    });
-    runOnUI(resetSwappingStatus)();
+      if (errorMessage) {
+        SwapInputController.quoteFetchingInterval.start();
 
-    if (errorMessage) {
-      SwapInputController.quoteFetchingInterval.start();
+        analyticsV2.track(analyticsV2.event.swapsFailed, {
+          createdAt: Date.now(),
+          type,
+          parameters,
+          selectedGas,
+          selectedGasSpeed,
+          slippage,
+          bridge: isBridge,
+          errorMessage,
+          inputNativeValue: SwapInputController.inputValues.value.inputNativeValue,
+          outputNativeValue: SwapInputController.inputValues.value.outputNativeValue,
+        });
+
+        if (errorMessage !== 'handled') {
+          logger.error(new RainbowError(`[getNonceAndPerformSwap]: Error executing swap: ${errorMessage}`));
+          const extractedError = errorMessage.split('[')[0];
+          Alert.alert(i18n.t(i18n.l.swap.error_executing_swap), extractedError);
+          return;
+        }
+      }
 
-      analyticsV2.track(analyticsV2.event.swapsFailed, {
+      queryClient.invalidateQueries([
+        // old user assets invalidation (will cause a re-fetch)
+        {
+          queryKey: userAssetsQueryKey({
+            address: parameters.quote.from,
+            currency: nativeCurrency,
+            connectedToHardhat,
+          }),
+        },
+        // new swaps user assets invalidations
+        {
+          queryKey: swapsUserAssetsQueryKey({
+            address: parameters.quote.from as Address,
+            currency: nativeCurrency,
+            testnetMode: !!connectedToHardhat,
+          }),
+        },
+      ]);
+
```

Comment:
@christianbaroni is there anything here with userAssetsStore that needs to be refetched/revalidated?